### ISSUE TITLE
Training on a single GPU

### ISSUE THREAD

**Issue by beezbo** [karpathy/llama2.c#221]
Has the example given for training on a single GPU in debug mode been tested to work ok to fully train the model? I want to try training the model since I haven't done it before, but not sure if the settings need to be tweaked to make sure the model turns out properly.

**Comment by richinseattle** [comment]
`train.py` will take about 20GB vRAM to train the default tiny 15M weight model.

**Comment by beezbo** [comment]
I think I have the vRAM covered. Have you run train.py in debug mode? Is it able to complete all the way through without needing to change the training parameters?

**Comment by richinseattle** [comment]
The debug run will not produce a working model. If you run train.py without any changes it will stop after 100,000 iterations and should be able to create a paragraph or two. The model will be saved to disk every 20,000 iterations if it has improved so you can test it as the training runs. I have produced models that trained for 100k & 200k iters on tinystories and now 500k iterations on wiki. The tinystories models can create basic coherent stories. The Wikipedia model creates things that look like Wikipedia excerpts but are completely hallucinated.

**Comment by beezbo** [comment]
The debug run will output checkpoints, but each checkpoint will be written over by the next one. I stopped it after it reached over 20k iterations. The models produce an output during inference, but I am not certain if it will continue to converge up to 100k iterations.

I also tried a train.py run in Kaggle and used 2 GPUs, but had to set the batch size to something like 60 to stay under the memory limit with some margin. Compile is set to false because it was having some compatibility issue and wouldn't run.

`!torchrun --standalone --nproc_per_node=2 train.py --compile=False --batch_size=60`

Does that look correct to being able to successfully train the model?

**Comment by beezbo** [comment]
This is what I have, but I modified the training script slightly. I would need to rerun it without mods if you are comparing it to the standard output. This is running on 2 GPUs, so it's not really relevant to my original question since I was wondering if debug mode could still properly train a model:

2400 | loss 1.5063 | lr 4.997533e-04 | 461.35ms | mfu 2.06%
2401 | loss 1.4834 | lr 4.997530e-04 | 459.82ms | mfu 2.06%
2402 | loss 1.4958 | lr 4.997526e-04 | 458.69ms | mfu 2.06%
2403 | loss 1.4984 | lr 4.997523e-04 | 459.70ms | mfu 2.07%
2404 | loss 1.4848 | lr 4.997519e-04 | 457.45ms | mfu 2.07%
2405 | loss 1.4913 | lr 4.997516e-04 | 458.32ms | mfu 2.07%
2406 | loss 1.5180 | lr 4.997512e-04 | 459.81ms | mfu 2.07%
2407 | loss 1.5500 | lr 4.997509e-04 | 459.08ms | mfu 2.07%
2408 | loss 1.4798 | lr 4.997505e-04 | 460.21ms | mfu 2.07%
2409 | loss 1.5128 | lr 4.997501e-04 | 458.27ms | mfu 2.07%
2410 | loss 1.5132 | lr 4.997498e-04 | 460.12ms | mfu 2.07%
2411 | loss 1.5215 | lr 4.997494e-04 | 457.54ms | mfu 2.07%
2412 | loss 1.5630 | lr 4.997491e-04 | 459.22ms | mfu 2.07%
2413 | loss 1.4953 | lr 4.997487e-04 | 458.48ms | mfu 2.07%
2414 | loss 1.5301 | lr 4.997484e-04 | 459.46ms | mfu 2.07%
